### PR TITLE
fix: prefix search

### LINKED ISSUE #3217: bug: user search select not displaying exact search results
when searching for a user in a large set of usernames that are similar, there's a chance you won't see the proper user in the dropdown.

In this instance, I'll use my own name as an example, we want to search for `Nathaniel.N`

but what shows are the following users:

- Random.Nathaniel
- Random.NathanielRandom
- NathanielRand.Koch
- NathanielRand.Smith
- NathanielRand.Weeks

### PR DESCRIPTION
**Pull request by andrewbenington**
- [x] Identified the issue which this PR solves.
- [x] Read the [**CONTRIBUTING**](https://github.com/target/goalert/blob/master/CONTRIBUTING.md) document.
- [x] Code builds clean without any errors or warnings.
- [x] Added appropriate tests for any new functionality.
- [x] All new and existing tests passed.
- [x] Added comments in the code, where necessary.
- [x] Ran `make check` to catch common errors. Fixed any that came up.

**Description:**

- Searching list pages (users, escalation policies, rotations, schedules, notifications, and services) only shows results that match the order of the search terms.
- Words can be skipped (e.g. the search "Serv Two" will return both "Service Two" and "Service One Two")

Before:
Screenshot 2023-08-17 at 12 57 45 PM
Screenshot 2023-08-17 at 12 57 35 PM

After:
Screenshot 2023-08-17 at 12 20 49 PM
Screenshot 2023-08-17 at 12 20 55 PM

**Which issue(s) this PR fixes:**
Fixes #3217 

**Describe any introduced API changes:**
Anything using `prefixSearch` now uses `orderedPrefixSearch`, which queries with a single regex.

Examples (`\m` matches beginning of word):
`\m{term 1}`
`\m{term 1}.*\m{term 2}`
`\m{term 1}.*\m{term 2}.*\m{term 3}`

### CODE REVIEW

File: search/render.go [hunk 1]
```diff
@@ -55,16 +55,26 @@ func splitRxTerms(rx string) pgtype.TextArray {
...
 		}
 
-		terms = append(terms, "\\m"+cur)
+		// prefix match terms with the \m "word start" symbol
+		if rx == "" {
+			rx = "\\m" + cur
+		} else {
+			// extra words in between are allowed with .*
+			rx = rx + ".*\\m" + cur
+		}
```
Comment (mastercactapus) [inline search/render.go:64]: Maybe a simpler way would be to change the original `splitRxTerms` func return `[]string` and then have a new `orderedPrefixRx` func call it and join with `.*`

mixing the join with the term separation is a little hard to follow

**Review by mastercactapus — APPROVED** [review]

**Review by tony-tvu — APPROVED** [review]
lgtm
